internal/app/go-filecoin/node/node.go

```diff
@@ -130,9 +130,6 @@ func (node *Node) Start(ctx context.Context) error {
 		return errors.Wrap(err, "failed to set up protocols:")
 	}
 
-	// DRAGONS: uncomment when we have retrieval market integration
```

Comment:
I looked at this not too long ago and it bothers me too. The same is happening with storage market; all the mining/market-related stuff is being started in one place.  I'm not sure what the decision making around this was and I haven't really had a chance to talk to anyone about it.

Reply:
It used to be the case that the network protocols weren't available until mining was started. The block mining submodule was probably being created here when the storage and retrieval markets were added, but I've fixed that so that the submodule is created at node construction and then augmented when mining starts.

Reply:
So reworking it is outside the scope of this PR, but it may need to be left where it is until this issue is resolved: https://github.com/filecoin-project/go-filecoin/issues/3137  . Retrieval is significantly simpler than storage so there's not much to do other than set up the APIs.